src/schoology-extractor/schoology_extractor/helpers/sync.py

```diff
@@ -40,29 +41,41 @@ def _resource_has_changed(
     db_engine:  sqlalchemy.engine.base.Engine,
     id_column: str = "id"
 ) -> bool:
-    query = f"SELECT * FROM {resource_name} where {id_column} = {resource[id_column]} limit 1;"
+    query = f"SELECT * FROM {resource_name} where SourceId = {resource[id_column]} limit 1;"
```

Comment:
`SourceId` will be the identifier from the LMS, it will be exposed in a column to simplify the process of querying existing resources to check if they have been updated.

Reply:
In at least one case - attendance events - the SourceId will be a string instead of an integer. This is because Schoology does not assign a source id for attendance. I created one with `f"{enrollment_id}#{event_date}`. I assume we will need to change the syntax here to have `... SourceId = '{resource[id_column]}' ... ` when we get to the story for setting sync dates on attendance events. Can you check if this will be a problem for integer values? I'm not asking to change it right now, as it is not necessary right now.

Reply:
I could define one argument for a "custom key", and use it when it is not null, so we could use it for specific resources.

Reply:
I made the change to set all SourceId columns to string(varchar, probably) in the sync database.